src/puppetlabs/trapperkeeper/services/authorization/authorization_core.clj

```diff
@@ -95,6 +81,18 @@
       (throw (IllegalArgumentException.
               (str "The authorization rule specified as " (pprint-rule rule)
                    " does not contain a '" (name k) "' key.")))))
+  (doseq [k [:sort-order :name]]
+    (when-not (get rule k)
+      (throw (IllegalArgumentException.
+              (str "The authorization rule specified as " (pprint-rule rule)
+                   " does not contain a '" (name k) "' key.")))))
+  (when (or (not (integer? (:sort-order rule)))
+            (< (:sort-order rule) 1)
+            (> (:sort-order rule) 999))
+    (throw (IllegalArgumentException.
+            (str "The sort-order set in the authorization rule specified as "
+                 (pprint-rule rule) " is invalid. It should be a number "
+                 "between 1-999."))))
```

Comment:
Maybe say "from 1 to 999" since "between" might imply that 1 and 999 are invalid?